core-plugins/src/main/java/io/cdap/plugin/batch/source/FileSourceConfig.java

```diff
@@ -82,6 +84,31 @@ public void validate(FailureCollector collector) {
       collector.addFailure("File system properties must be a valid json.", null)
         .withConfigProperty(NAME_FILE_SYSTEM_PROPERTIES).withStacktrace(e.getStackTrace());
     }
+    validateSchemaDelimitedFormats(collector);
+  }
+
+  public void validateSchemaDelimitedFormats(FailureCollector collector) {
+    if (getSchema() == null) {
+      return;
+    }
+    if (getFormatName().equalsIgnoreCase(FileFormat.CSV.name())
+      || getFormatName().equalsIgnoreCase(FileFormat.TSV.name())
+      || getFormatName().equalsIgnoreCase(FileFormat.DELIMITED.name())) {
+      for (Schema.Field field : getSchema().getFields()) {
+        Schema.LogicalType type = field.getSchema().getLogicalType();
+        if (type == null) {
+          continue;
+        }
+        if (type.equals(Schema.LogicalType.TIME_MICROS) || type.equals(Schema.LogicalType.TIME_MILLIS)
+        || type.equals(Schema.LogicalType.DATE)) {
+          collector.addFailure(
+            String.format("Schema containing 'datetime' or 'time' fields is not supported for %s format.",
+                          getFormatName()),
+            "Change type to 'string'."
```

Comment:
We can write a more cleaner message for the user in here. 

We should also remove hard coded '`datetime`', '`time`' and use `type` instead to catch the specific schema type. 

In corrective action, we should also mention the `type` that user should change to string.

Reply:
PR updated.